src/main/java/sirius/web/security/ScopeInfo.java

```diff
@@ -281,28 +296,38 @@ private void fillConfig(Object result) {
                                                                     f.getAnnotation(HelperConfig.class).value()));
     }
 
-    private void fillFriends(Object result) {
+    private void fillFriends(Object result, int circuitBreaker) {
         Reflection.getAllFields(result.getClass())
                   .stream()
-                  .filter(f -> f.isAnnotationPresent(Helper.class))
-                  .forEach(f -> {
+                  .filter(field -> field.isAnnotationPresent(Helper.class))
+                  .forEach(field -> {
                       try {
-                          f.setAccessible(true);
-                          f.set(result, makeHelperByType(f.getType()));
+                          fillFriend(result, field, circuitBreaker);
                       } catch (Exception e) {
                           Exceptions.handle()
                                     .error(e)
                                     .to(UserContext.LOG)
                                     .withSystemErrorMessage("Cannot fill friend %s in %s of helper %s (%s): %s (%s)",
-                                                            f.getType().getName(),
-                                                            f.getName(),
+                                                            field.getType().getName(),
+                                                            field.getName(),
                                                             result,
                                                             result.getClass().getName())
                                     .handle();
                       }
                   });
     }
 
+    private void fillFriend(Object result, Field field, int circuitBreaker) throws IllegalAccessException {
+        HelperFactory<?> factory = findFactoryForType(field.getType());
+
+        if (circuitBreaker > HELPER_FRIENDS_MAX_DEPTH) {
+            throw new IllegalStateException("Detected circular friend dependency");
```

Comment:
maybe add the field / result / result.getClass() in the message so simplify error tracing...